### ISSUE TITLE
`test/fulfillment` has 5/? assertions passing. Fix indexing related issue

### ISSUE THREAD

**Issue by Connoropolous** [holo-rea/holo-rea#252]
- [x] fix indexing related issue

```
cd test
WASM_LOG=debug RUST_LOG=error RUST_BACKTRACE=1 npx tape fulfillment/**/*.js
```

```
not ok 6 EconomicEvent.fulfills value present
  ---
    operator: ok
    expected: true
    actual:   undefined
```

for this object:
```
{
  economicEvent: {
    id: [
      <Buffer ...>,
      <Buffer ...>
    ],
    revisionId: <Buffer ...>,
    action: 'raise',
    note: 'test event which is fulfilling a commitment',
    provider: [
      <Buffer ...>,
      <Buffer ...>
    ],
    receiver: [
      <Buffer ...>,
      <Buffer ...>
    ],
    resourceClassifiedAs: [ 'some-resource-type' ],
    resourceQuantity: { hasNumericalValue: 1, hasUnit: [Array] },
    hasPointInTime: '2019-11-19T04:29:55.056+00:00'
  }
}
```

related code:
https://github.com/h-REA/hREA/blob/a44147d0a5568a472fcbab2432c0c4cd2ca631d9/test/fulfillment/fulfillment_records_e2e.js#L61-L63

**Comment by Connoropolous** [comment]
Output: 

```
18:26:30 [tryorama] info: Using the following settings from environment variables:
18:26:30 [tryorama] info: {
  "adminInterfaceId": "tryorama-interface-admin",
  "appInterfaceId": "tryorama-interface-app",
  "stateDumpOnError": true,
  "zomeCallTimeoutMs": 90000,
  "conductorTimeoutMs": 125000,
  "strictConductorTimeout": false,
  "chooseFreePort": false,
  "portRange": [
    33000,
    34000
  ],
  "legacy": false,
  "singletonAppId": "TRYORAMA_APP",
  "holochainPath": "holochain",
  "lairPath": "lair-keystore"
}
18:26:30 [tryorama] debug: About to execute 1 tests
18:26:30 [tryorama] debug: Executing test: links can be written and read between DNAs
TAP version 13
# links can be written and read between DNAs
18:26:30 [tryorama] debug: api.players: creating players
18:26:30 [tryorama] debug: api.players: seed args generated for c0 = {"configDir":"/var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5","adminInterfacePort":33000,"scenarioName":"links can be written and read between DNAs","playerName":"c0","uid":"eb37a55c-c142-43b4-aacd-3fa79079dbdb"}
18:26:30 [tryorama] debug: built config: {"network":{"network_type":"quic_bootstrap","transport_pool":[{"type":"quic"}]},"environment_path":"/var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5","keystore":{"type":"lair_server_legacy_deprecated","keystore_path":"/var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5/keystore","danger_passphrase_insecure_from_config":"pass"},"admin_interfaces":[{"driver":{"type":"websocket","port":33000}}],"app_interfaces":[{"driver":{"type":"websocket","port":0}}]}
18:26:30 [tryorama] debug: api.players: player config committed for c0
18:26:30 [tryorama] debug: api.players: players built
FIXME: ignoring onJoin
18:26:30 [tryorama: player c0] debug: starting up
18:26:30 [tryorama] info: Spawning lair for test with keystore at:  /var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5/keystore
18:26:31 [tryorama] info: Using conductor path:  holochain
18:26:31 [tryorama] info: Holochain version:     holochain 0.0.128

18:26:31 [tryorama] info: Conductor config path: /var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5/conductor-config.yml
18:26:31 [tryorama] debug: running: holochain -c /var/folders/5s/rldbq43s2yj4v5xb89bwygv80000gn/T/tryorama/EhhQx5/conductor-config.yml
18:26:31 info:
☉☉☉ [[[CONDUCTOR c0]]]
☉
☉ ###HOLOCHAIN_SETUP###
☉ ###ADMIN_PORT:33000###
☉ ###HOLOCHAIN_SETUP_END###
☉ Conductor ready.
☉
18:26:31 [tryorama] info: Conductor 'c0' process spawning completed.
18:26:31 [tryorama: tryorama conductor c0] debug: Conductor constructing
18:26:31 [tryorama: player c0] debug: started up
18:26:31 [tryorama: player c0] debug: initializing
18:26:31 [tryorama: tryorama conductor c0] debug: connectInterfaces :: connected admin interface at ws://localhost:33000
App Port spun up on port  52017
18:26:31 [tryorama: tryorama conductor c0] debug: connectInterfaces :: connected app interface at ws://localhost:52017
18:26:31 [tryorama: player c0] debug: initialized
18:26:31 [tryorama: player c0] debug: Player.installHapps
18:26:31 [tryorama: player c0] debug: Player.adminWs()
18:26:31 [tryorama: player c0] debug: Player.installHapp(["/Users/connor/code/hrea/bundles/dna/planning/hrea_planning.dna","/Users/connor/code/hrea/bundles/dna/observation/hrea_observation.dna"], noAgentPubKey)
18:26:44 [tryorama: player c0] debug: Player.adminWs()
18:26:44 [tryorama: player c0] debug: Player.appWs()
Init Holochain connection: ws://localhost:52017
Holochain connection to ws://localhost:52017 OK
18:26:44 [tryorama: player c0] debug: cell.call()
ok 1 commitment created successfully
18:26:44 [tryorama: player c0] debug: cell.call()
ok 2 event created successfully
18:26:44 [tryorama: player c0] debug: cell.call()
18:26:44 [tryorama] error: Test error: {
  type: 'error',
  data: {
    type: 'ribosome_error',
    data: `Wasm error while working with Ribosome: Guest("Host(\\"An error with entry defs in zome 'fulfillment': entry def not found for App(\\\\\\"FulfillmentAddress\\\\\\")\\")")`
  }
}
not ok 3 Test threw an exception. See output for details.
  ---
    operator: fail
    at: <anonymous> (/Users/connor/code/hrea/node_modules/.pnpm/@holochain+tryorama@0.4.10/node_modules/@holochain/tryorama/src/middleware.ts:137:13)
    stack: |-
      Error: Test threw an exception. See output for details.
          at Test.assert [as _assert] (/Users/connor/code/hrea/node_modules/.pnpm/tape@4.15.0/node_modules/tape/lib/test.js:275:54)
          at Test.bound [as _assert] (/Users/connor/code/hrea/node_modules/.pnpm/tape@4.15.0/node_modules/tape/lib/test.js:89:32)
          at Test.fail (/Users/connor/code/hrea/node_modules/.pnpm/tape@4.15.0/node_modules/tape/lib/test.js:368:10)
          at Test.bound [as fail] (/Users/connor/code/hrea/node_modules/.pnpm/tape@4.15.0/node_modules/tape/lib/test.js:89:32)
          at /Users/connor/code/hrea/node_modules/.pnpm/@holochain+tryorama@0.4.10/node_modules/@holochain/tryorama/src/middleware.ts:137:13
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
  ...
18:26:44 [tryorama] debug: Test failed: links can be written and read between DNAs {
  type: 'error',
  data: {
    type: 'ribosome_error',
    data: `Wasm error while working with Ribosome: Guest("Host(\\"An error with entry defs in zome 'fulfillment': entry def not found for App(\\\\\\"FulfillmentAddress\\\\\\")\\")")`
  }
}
18:26:44 [tryorama] debug: Cleaning up test: links can be written and read between DNAs
18:26:44 [tryorama] debug: Calling Api._cleanup. description: links can be written and read between DNAs
18:26:44 [tryorama: player c0] debug: calling Player.cleanup, conductor: %b
18:26:44 [tryorama: player c0] debug: Shutting down...
18:26:44 [tryorama: tryorama conductor c0] debug: Killing...
18:26:44 [tryorama] info: conductor 'c0' exited with code null
18:26:44 [tryorama: player c0] debug: Shut down.
FIXME: ignoring onLeave
18:26:44 [tryorama] debug: Finished with test: links can be written and read between DNAs

1..3
# tests 3
# pass  2
# fail  1

**Comment by pospi** [comment]
Fixed in 08de190c
